File: 문제풀이/chap6-16_20/prob6-17.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 입력
		Scanner sc = new Scanner(System.in);
		long N = sc.nextLong();
		
		// 세로 길이를 1부터 √N까지 전체 탐색
		long answer = (1L << 60);
		for (long x = 1; x * x <= N; x++) {
			if (N % x == 0) {
				answer = Math.min(answer, 2 * x + 2 * (N / x));
			}
		}
		
		// 답 출력
		System.out.println(answer);
	}
}

File: 문제풀이/chap6-16_20/prob6-17.py
# 입력
N = int(input())

# 세로 길이를 1부터 √N까지 전체 탐색
LIMIT = int(N ** 0.5)
answer = 10 ** 19
for i in range(1, LIMIT + 1):
	if N % i == 0:
		answer = min(answer, 2 * i + 2 * (N // i))

# 답 출력
print(answer)